File: python/pages/checkout_page.py
from playwright.sync_api import Page


class CheckoutPage:
	"""POM for Checkout page (synchronous Playwright).

	Helpers to proceed to checkout, fill payment details, and place the order.
	"""

	def __init__(self, page: Page):
		self.page = page

	def proceed_to_checkout(self) -> None:
		self.page.click("#checkout")

	def fill_payment_details(self, card: str, expiry: str, cvv: str) -> None:
		self.page.fill("#cardNumber", card)
		self.page.fill("#expiry", expiry)
		self.page.fill("#cvv", cvv)

	def place_order(self) -> None:
		self.page.click("#placeOrder")

File: js/pages/checkoutPage.ts
import { Page } from '@playwright/test';

// POM for Checkout page
export class CheckoutPage {
  constructor(private page: Page) { }

  // Proceed to checkout from cart
  async proceedToCheckout() {
    await this.page.click('#checkout');
  }

  // Fill payment details (dummy test card values)
  async fillPaymentDetails(card: string, expiry: string, cvv: string) {
    await this.page.fill('#cardNumber', card);
    await this.page.fill('#expiry', expiry);
    await this.page.fill('#cvv', cvv);
  }

  // Place the order
  async placeOrder() {
    await this.page.click('#placeOrder');
  }
}